syntax = "proto3";

package policy;

service ClientService {
    rpc RegisterClient (ClientDetails) returns (ClientResponse);
    rpc GetClientDetails (ClientRequest) returns (GetClientDetailsResponse);
    rpc GetPolicy (PolicyRequest) returns (PolicyResponse);
}

service LogService {  // NEW SERVICE for logging
    rpc SendLog (LogRequest) returns (LogResponse);
}

// Client Registration
message ClientRequest {
    string client_id = 1;
    string agent_id = 2;
}

message ClientDetails {
    string device_name = 1;
    string os_version = 2;
    string username = 3;
}

message ClientResponse {
    string status = 1;
    string message = 2;
    string client_id = 3;
    string agent_id = 4;
}

message GetClientDetailsResponse {
    string device_name = 1;
    string os_version = 2;
    string username = 3;
}

// Policy Retrieval
message PolicyRequest {
    string client_id = 1;
    string agent_id = 2;
}

message PolicyResponse {
    string status = 1;
    string message = 2;
    string policy_data = 3; // JSON policy as a string
}

// Logging
message LogRequest {
    string client_id = 1;
    string agent_id = 2;
    string log_level = 3; // INFO, WARNING, ERROR, etc.
    string message = 4;
}

message LogResponse {
    string status = 1;
    string message = 2;
}
